src/java.base/share/classes/java/io/ObjectInputFilter.java

```diff
@@ -48,50 +57,163 @@
  * practices for defensive use of serial filters.
  * </strong></p>
  *
- * If set on an {@link ObjectInputStream}, the {@link #checkInput checkInput(FilterInfo)}
- * method is called to validate classes, the length of each array,
+ * <p>To protect against deserialization vulnerabilities, application developers
+ * need a clear description of the objects that can be deserialized
+ * by each component or library. For each context and use case, developers should
+ * construct and apply an appropriate filter.
+ *
+ * <p>For simple cases, a static filter can be {@linkplain Config#setSerialFilter(ObjectInputFilter) set}
+ * for the entire application.
+ * For example, a filter that allows example classes, allows classes in the
+ * {@code java.base} module, and rejects all other classes:
+ *
+ * <pre>{@code
+ *     var filter = ObjectInputFilter.Config.createFilter("example.*;java.base/*;!*")
+ *     ObjectInputFilter.Config.setSerialFilter(filter);
+ * }</pre>
```

Comment:
It's good to have a straightforward example, but it has an implicit assumption - that the built-in filter factory is in operation ( and will not change ). I wonder if there is a way to update the example (without too much fuss), or otherwise add a textual qualification. Though, I'm not sure what this would look like.